server/src/main/java/org/opensearch/cluster/routing/allocation/allocator/BalancedShardsAllocator.java

```diff
@@ -139,6 +139,14 @@ public class BalancedShardsAllocator implements ShardsAllocator {
         Property.NodeScope
     );
 
+    public static final Setting<Long> PRIMARY_CONSTRAINT_THRESHOLD_SETTING = Setting.longSetting(
+        "cluster.routing.allocation.primary_constraint.threshold",
+        10,
+        0,
+        Property.Dynamic,
+        Property.NodeScope
+    );
```

Comment:
I am assuming default will be set as 10?

Reply:
Yes